tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/MuleClassPathClassifier.java

```diff
@@ -0,0 +1,394 @@
+                                       )).resolveDependencies().stream().filter(d -> !d.isPomType()).map(dependency -> artifactToClassPathURLResolver.resolveURL(dependency, classPathURLs)).forEach(extensionURLs::add);
+
+        if (extensionURLs.size() == sizeBeforeDepResolver)
+        {
+            throw new IllegalStateException("There should be at least one compile dependency found that matched to extension: '" + extension.getName() +
+                                            "'. Be aware that compile scope is what the classification uses for selecting the urls to be added to plugin class loader");
+        }
+        return new ExtensionPluginUrlClassification(extension, extensionURLs);
+    }
+
+    /**
+     * Does the classification where the rootArtifact is not an extension so if it is not an extension it is treated as a plugin
+     * where its compile dependencies and transitive dependencies should go to a plugin {@link ClassLoader}.
+     *
+     * @param exclusion
+     * @param artifactToClassPathURLResolver
+     * @param allDependencies
+     * @param compileArtifact
+     * @param targetTestClassesFolder
+     * @param classPathURLs
+     * @return
+     */
+    private PluginUrlClassification pluginClassPathClassification(final Predicate<MavenArtifact> exclusion, final MavenArtifactToClassPathURLResolver artifactToClassPathURLResolver, final LinkedHashMap<MavenArtifact, Set<MavenArtifact>> allDependencies, final MavenArtifact compileArtifact, final File targetTestClassesFolder, final List<URL> classPathURLs)
+    {
+        Set<URL> urls = new DependencyResolver(new ConfigurationBuilder()
+                                       .setMavenDependencyGraph(allDependencies)
+                                       .includeRootArtifact()
+                                       .selectDependencies(
+                                               new DependenciesFilterBuilder()
+                                                       .match(dependency -> dependency.isCompileScope() && !exclusion.test(dependency))
+                                       )
+                                       .collectTransitiveDependencies(
+                                               new TransitiveDependenciesFilterBuilder()
+                                                       .match(transitiveDependency -> transitiveDependency.isCompileScope() && !exclusion.test(transitiveDependency))
+                                       )).resolveDependencies().stream().filter(d -> !d.isPomType()).map(dependency -> artifactToClassPathURLResolver.resolveURL(dependency, classPathURLs)).collect(Collectors.toSet());
+        return new PluginUrlClassification(compileArtifact.getArtifactId(), Lists.newArrayList(urls));
+    }
+
+    /**
+     * The list of exclusion GAT to be excluded from application/plugin classloaders due to these are supposed to
```

Comment:
GroupId,ArtifactId,Type, changed to explicitly mention the complete word instead.